src/dense.jl

```diff
@@ -48,12 +48,28 @@ end
 
 Base.length(dense::DenseOutput) = length(dense.opts.tout)
 
+tdir(ivp, solver::DenseOutput) = sign(solver.opts.tstop - ivp.t0)
+
 @compat function (::Type{DenseOutput{I}}){T,I}(ivp::IVP{T};
+                                               tstop = T(Inf),
+                                               tout::AbstractVector{T} = T[tstop],
                                                opts...)
+    if all(tout.>=ivp.t0)
+        tout = sort(tout)
```

Comment:
With `sort` we don't have to restrict ourselves to any of these, although now that I think about it, if we sort `tout` we loose the 1:1 relationship with the order of `tout` the user provided.